src/store/modules/user.js

```diff
@@ -0,0 +1,178 @@
+
+import api from "../api"
+import * as types from "../mutation-types"
+
+export const CART_KEY = 'sell-media-cart'
+export const LIGHTBOX_KEY = 'sell-media-lightbox'
+export const USAGE_KEY = 'sell-media-usage'
+
+// State: the state properties for the module
+const state = {
+  user: null,
+  cart: JSON.parse(window.localStorage.getItem(CART_KEY) || '[]'),
+  lightbox: JSON.parse(window.localStorage.getItem(LIGHTBOX_KEY) || '[]'),
+  usage: JSON.parse(window.localStorage.getItem(USAGE_KEY) || '[]')
+}
+
+// Getters: Used to retrieve state properties
+const getters = {
+  user: state => state.user,
+  lightbox: state => state.lightbox,
+  cart: state => state.cart,
+  usage: state => state.usage
+}
+
+// Actions: Used to trigger asyncronous tasks and commit mutations
+const actions = {
+  /**
+   * Sets the user to the WP user object
+   * @param { user } the user object
+   */
+  setUser({ commit, state }, user) {
+    commit(types.SET_USER, user)
+  },
+
+  /**
+   * Adds a given product to the users lightbox.
+   * @param { product } the product to be added
+   */
+  addToLightbox ({ commit, state, getters }, product) {
+    let newLightbox = state.lightbox
+    newLightbox.push(product)
+    commit(types.SET_LIGHTBOX, newLightbox)
+  },
+
+  /**
+   * Removes a given product from the users lightbox.
+   * @param { product } the product to be removed
+   */
+  removeFromLightbox ({ commit, state, getters }, product) {
+    let newLightbox = state.lightbox
+    newLightbox.splice(newLightbox.indexOf(product), 1)
+    commit(types.SET_LIGHTBOX, newLightbox)
+  },
+
+  /**
+   * Resets the lightbox
+   */
+  deleteLightbox ({ commit }) {
+    commit(types.SET_LIGHTBOX, JSON.parse('[]'))
+  },
+
+  /**
+   * Adds a product object to the cart
+   * @param { product } the product to be added
+   */
+  addToCart ({ commit }, product) {
+    commit(types.ADD_TO_CART, product)
+  },
+
+  /**
+   * Removes a product object to the cart
+   * @param { product } the product to be removed
+   */
+  removeFromCart ({ commit }, product) {
+    commit(types.REMOVE_FROM_CART, product)
+  },
+
+  /**
+   * Replaces an existing product object with an updated version
+   * @param { product } the product to be updated
+   */
+  updateCartProduct ({ commit }, product) {
+    commit(types.UPDATE_CART_PRODUCT, product)
+  },
+
+  /**
+   * Resets the cart state
+   */
+  deleteCart ({ commit }) {
+    commit(types.DELETE_CART)
+  },
+
+  /**
+   * Sets the usage/license for the user
+   * @param { value } the usage value
+   */
+  setUsage ({ commit }, value) {
+    commit(types.SET_USAGE, value)
+  },
+
+  /**
+   * Resets the usage for the user
+   */
+  deleteUsage ({ commit }) {
+    commit(types.SET_USAGE, JSON.parse('[]'))
+  },
+
+  // TODO: This mutation was called from app.js but wasn't defined. @EvanAgee
+  initCart ({ state }) {
+    return true
+  },
+
+  // TODO: This isn't being referenced anywhere, please advise. @EvanAgee
```

Comment:
This can be removed @EvanAgee . We are verifying product prices/licenses/discounts server side (PHP) right after Stripe returns it's token and before we charge the buyer. This allows us to charge the buyer the correct price even if they modify the prices in localstorage.